shotover-proxy/src/transforms/distributed/consistent_scatter.rs

```diff
@@ -52,49 +49,36 @@ impl ConsistentScatterConfig {
     }
 }
 
-fn get_timestamp(frag: &QueryResponse) -> i64 {
-    debug!("\n\n {:#?} \n\n", frag.response_meta);
-    if let Some(MessageValue::Document(meta)) = frag.response_meta.as_ref() {
-        if let Some(t) = meta.get("timestamp") {
-            if let MessageValue::Integer(i, _) = t {
-                return *i;
-            }
-            return 0;
-        }
-    }
-    0
-}
-
-fn get_size(frag: &QueryResponse) -> usize {
-    frag.result.as_ref().map_or(0, std::mem::size_of_val)
+fn get_size(_message: &mut Message) -> usize {
+    4 // TODO: Implement. Old impl was just removed because it was broken anyway
 }
 
-fn resolve_fragments(fragments: &mut Vec<QueryResponse>) -> Option<QueryResponse> {
-    let mut newest_fragment: Option<QueryResponse> = None;
-    let mut biggest_fragment: Option<QueryResponse> = None;
+fn resolve_fragments(fragments: &mut Vec<Message>) -> Option<Message> {
+    let mut newest_fragment: Option<Message> = None;
+    let mut biggest_fragment: Option<Message> = None;
 
     // Check the age of the response, store most recent
     // If we don't have an age, store the biggest one.
     // Return newest, otherwise biggest. Returns newest, even
     // if we have a bigger response.
     while !fragments.is_empty() {
-        if let Some(fragment) = fragments.pop() {
-            let candidate = get_timestamp(&fragment);
+        if let Some(mut fragment) = fragments.pop() {
+            let candidate = fragment.meta_timestamp.unwrap_or(0);
             if candidate > 0 {
-                match newest_fragment {
+                match &mut newest_fragment {
```

Comment:
Im surprised clippy isnt complaining about it but this can be just `&` instead of `&mut`